lib/Service/TableService.php

```diff
@@ -180,6 +180,28 @@ public function find(int $id, ?string $userId = null, bool $skipTableEnhancement
 		}
 	}
 
+	/**
+	 * @param string $term
+	 * @param int $limit
+	 * @param int $offset
+	 * @param string|null $userId
+	 * @return array
+	 */
+	public function search(string $term, int $limit = 100, int $offset = 0, ?string $userId = null): array {
+		try {
+			/** @var string $userId */
+			$this->permissionsService->preCheckUserId($userId);
+			$tables = $this->mapper->search($term, $userId, $limit, $offset);
+			foreach ($tables as &$table) {
+				/** @var string $userId */
+				$this->enhanceTable($table, $userId);
```

Comment:
Same here, IMO no need to redefine the type.
Also, `$userId` can be null there. I would add a check or adjust `enhanceTable`.
Or maybe `InternalError` is thrown in this case?